spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaMessageListenerContainerTests.java

```diff
@@ -515,6 +518,7 @@ public void testRecordAck() throws Exception {
 			finally {
 				for (Entry<TopicPartition, OffsetAndMetadata> entry : map.entrySet()) {
 					if (entry.getValue().offset() == 2) {
+						recordTime.add(System.currentTimeMillis());
```

Comment:
I think you need to stub `poll()` instead (and `callRealMethod()`); there is no guarantee that there will be 2 calls here - all 4 records might be returned by the first `poll()`.

Reply:
Well, that's why I added that `props.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1);` config above.

Reply:
Doh; sorry.